File: Advanced/PageAndRDL/CustomPdfExport/VB.NET/PdfRendering/Rendering/PdfContentGenerator.Font.vb
Imports PdfSharp.Drawing
Imports GrapeCity.ActiveReports.Drawing

Partial Class PdfContentGenerator
	Private Class Font
		Private ReadOnly _fontInfo As XFont

		Public Sub New(fontInfo As XFont)
			_fontInfo = fontInfo
		End Sub

		Public Function Clone() As Object
			Return New Font(_fontInfo)
		End Function

		Public Shared Widening Operator CType(font As Font) As XFont
			Return font._fontInfo
		End Operator
	End Class
End Class

File: Advanced/PageAndRDL/CustomPdfExport/C#/PdfRendering/Rendering/PdfContentGenerator.Font.cs
using GrapeCity.ActiveReports.Drawing;
using PdfSharp.Drawing;

namespace GrapeCity.ActiveReports.Samples.Export.Rendering.PdfSharp
{
	partial class PdfContentGenerator
	{
		private class Font
		{
			private readonly XFont _fontInfo;

			public Font(XFont fontInfo)
			{
				_fontInfo = fontInfo;
			}
			
			public object Clone()
			{
				return new Font(_fontInfo);
			}

			public static implicit operator XFont(Font font)
			{
				return font._fontInfo;
			}
		}
	}
}
